### DISCUSSION TITLE
How to user pmd-xml to analyse pom.xml?

### DISCUSSION THREAD

**Discussion by luckyDOT** [pmd/pmd#4143] [Q&A]
I am trying to expand pmd to analyse pom.xml recently,but it actually not work,it seems to work on java file.I use XmlXPathRule class,who can give a advice?

**Comment by jsotuyod** [comment]
Sorry I'm seeing this so late.

Pmd already has support to analyze pom files https://pmd.github.io/latest/pmd_rules_pom.html

So I'm unsure what you are trying to achieve.

**Comment by adangel** [comment]
You are probably suffering from this problem: #1540
pom.xml files have the extension "xml" and not "pom"... but the extension "xml" is already used by the general xml language module.

Since PMD 6.37.0 there is a workaround with `--force-language` and the following execution indeed works:

```shell
pmd-bin-6.50.0/bin/run.sh pmd -d pom.xml -f text -R category/pom/errorprone.xml/ProjectVersionAsDependencyVersion --force-language pom
```

You'll need to mention the pom.xml files explicitly (just giving a directory doesn't work as PMD would search for *.pom files...) and add the "--force-language pom" option.
